demos/classification_demo/main.cpp

```diff
@@ -0,0 +1,537 @@
+                    }
+
+                    shownImagesInfo.push_back(
+                        LabeledImage{completedInferRequestInfo->images[i].mat, predictedLabels[i], predictionResult});
+                }
+
+                framesNum += FLAGS_b;
+                
+                avgFPS = framesNum / std::chrono::duration_cast<Sec>(
+                    std::chrono::steady_clock::now() - startTime).count();
+                gridMat.updateMat(shownImagesInfo);
+                auto processingEndTime = std::chrono::steady_clock::now();
+                for (const auto & image : completedInferRequestInfo->images) {
+                    latencySum += processingEndTime - image.startTime;
+                }
+                avgLatency = std::chrono::duration_cast<Sec>(latencySum).count() / framesNum;
+                accuracy = static_cast<double>(correctPredictionsCount) / framesNum;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, !FLAGS_gt.empty(), presenter);
+                
+                if (!FLAGS_no_show) {
+                    cv::imshow("classification_demo", gridMat.outImg);
+                    key = static_cast<char>(cv::waitKey(1));
+                    presenter.handleKey(key);
+                }
+            } else if (!emptyInferRequests.empty()) {
+                auto inferRequestStartTime = std::chrono::steady_clock::now();
+                cv::Mat nextImage = resizeImage(inputImages[nextImageIndex], modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                                                    {nextImage,
+                                                     classIndices[nextImageIndex],
+                                                     inferRequestStartTime});
+                nextImageIndex++;
+                if (nextImageIndex == imageNames.size()) {
+                    nextImageIndex = 0;
+                }
+                if (emptyInferRequests.front().images.size() == FLAGS_b) {
+                    auto emptyInferRequest = emptyInferRequests.front();
+                    emptyInferRequests.pop();
+
+                    emptyInferRequest.inferRequest.SetCompletionCallback([emptyInferRequest,
+                                                      &completedInferRequests,
+                                                      &mutex,
+                                                      &condVar,
+                                                      &hasCallbackException,
+                                                      &irCallbackException] {
+                        {
+                            std::lock_guard<std::mutex> callbackLock(mutex);
+                        
+                            try {
+                                completedInferRequests.push(emptyInferRequest);
+                            }
+                            catch(...) {
+                                if (!hasCallbackException) {
```

Comment:
As long as you have that lock, `hasCallbackException` is redundant. You could just use `if (!irCallbackException)` instead.